### ISSUE TITLE
Screen readers focus position resets when going back

### ISSUE THREAD

**Issue by Foohx** [react-navigation/react-navigation#11189]
### Current behavior

I am currently working on the accessibility of an application for visually impaired people and I am having a focus problem with screen readers (VoiceOver and TalkBack).

When a user activates an interactive element (Button, TouchableOpacity, ...) to navigate to another screen and then decides to return to the previous screen, the screen reader's focus is reset to the first element.

Below is an example of what happens:

https://user-images.githubusercontent.com/7073407/216574419-a83f9e9e-3401-4b87-a299-a827948172ef.mp4

### Expected behavior

Normally when a user returns to the previous screen the focus should be on the last item they interacted with.

Imagine the pain these users experience when they open an item in a FlatList. They have to go through everything again.

### Reproduction

https://snack.expo.dev/@foohx/reset-sr-focus-when-going-back

### Platform

- [X] Android
- [X] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [X] @react-navigation/bottom-tabs
- [ ] @react-navigation/drawer
- [ ] @react-navigation/material-bottom-tabs
- [ ] @react-navigation/material-top-tabs
- [X] @react-navigation/stack
- [X] @react-navigation/native-stack
- [ ] react-native-tab-view
- [ ] flipper-plugin-react-navigation

### Environment

I've removed the packages that I don't use

| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/native               |v6.1.2
| @react-navigation/bottom-tabs          |v6.5.3
| @react-navigation/native-stack         |v6.9.8
| react-native-safe-area-context         |v4.4.1
| react-native-screens                   | v3.18.0
| expo                                   |v47.0.0

**Comment by Foohx** [comment]
Issue still here after upgrading to the latest versions.

**Comment by scottgrunerud** [comment]
Any movement on this we are also having the same issue, in the latest version and Expo 51. @Foohx did you figure out a work around ?

**Comment by Foohx** [comment]
> Any movement on this we are also having the same issue, in the latest version and Expo 51. @Foohx did you figure out a work around ?

Hi @scottgrunerud, 

Unfortunately, no! The issue is still present, and I have no alternative solutions! 

A workaround could have been to use `setAccessibilityFocus()` with screen focus state and listen to the focus on each element of each screen, but for me `setAccessibilityFocus()` seems to be inconsistent.

This is starting to become problematic.

**Comment by scottgrunerud** [comment]
Yeah, seems like there is a lack of consistency on how each operating system handles the focus. If you checkout the settings page for an Android device it will restore focus back to the last element, but on iOS it always goes to the top.

**Comment by Foohx** [comment]
So indeed, on iOS, the focus is reset to the top only when using the back button. However, if we perform an Escape Gesture (the letter Z with two fingers), in that case, the focus is managed correctly like on Android. In React Navigation, the focus is always reset to the top regardless of the return method.

**Comment by buschco** [comment]
> Normally when a user returns to the previous screen the focus should be on the last item they interacted with.

I do not think it always behaves like that. In Apple Health for example the focus is starts at the top again.

**Comment by GeneralKFC** [comment]
Hi! Is this problem solved yet? And if so, are there any links to the solution?

**Comment by izyspania** [comment]
I am looking for a solution too

**Comment by ArturKalach** [comment]
Hello there,

> > Normally when a user returns to the previous screen the focus should be on the last item they interacted with.
> 
> I do not think it always behaves like that. In Apple Health for example the focus is starts at the top again.

I agree with this, it doesn’t always work this way. For some apps, like the App Store, it’s implemented, but for others, like Notes and Photos, it’s not. In some apps, you might also notice that even after closing a modal, the focus isn’t returned to the component that triggered it.

Basically, Apple provides an API to implement this, so I think it should be possible to add a prop and handle it.

_____

Recently, I created a component for this case, but it might be better to add this feature to `react-navigation`. If I have some time, I’ll try to do it.

```tsx
import { A11y } from 'react-native-a11y-order';

const HomeScreen = () =>  {
  const isFocused = useIsFocused();
  return (
    <View>
      <A11y.ScreenChange
        title="Home Screen"
        displayed={isFocused}
      />
      ....
    </View>
)
}
```

It tracks focus changes from the native side and stores them for the UIViewController. I think it’s also possible to create a patch or add a native mixin to solve this globally.
